x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -191,12 +202,25 @@ private void searchGenericEvents(GetStackTracesRequest request, ActionListener<G
                     .field(request.getStackTraceIds())
             )
             .execute(handleEventsGroupedByStackTrace(client, responseBuilder, submitListener, searchResponse -> {
+                long totalSamples = 0;
                 StringTerms stacktraces = searchResponse.getAggregations().get("group_by");
-                Map<String, Long> countPerStackTrace = new TreeMap<>();
-                for (StringTerms.Bucket bucket : stacktraces.getBuckets()) {
-                    countPerStackTrace.put(bucket.getKeyAsString(), bucket.getDocCount());
+                List<HostEventCount> hostEventCounts = new ArrayList<>(102400); // Preparation for future host.id aggregation
```

Comment:
How did you come up with that number?

Reply:
I made it a constant with an explanation.